### PR TITLE
cmd ref: add a note about SSH/SFTP quirks in remote add

### LINKED ISSUE #781: remote: add a note that SFTP support is required for ssh
https://github.com/iterative/dvc/issues/2755#issuecomment-552547440

### LINKED ISSUE #802: remote: ssh: add a note about sftp root
We've seen a few users (https://github.com/iterative/dvc/issues/2755#issuecomment-555385560, https://github.com/iterative/dvc/issues/2535) getting confused by their sftp server being configured with a root directory that differs from a physical root. We will try to find a programmatic way to detect that https://github.com/iterative/dvc/issues/2812 , but we should document it none the less.

### PR DESCRIPTION
**Pull request by efiop**
Fixes #781 
Fixes #802

**IMPORTANT NOTES** (please read, then delete):

- Have you followed the guidelines in
  [Contributing Documentation](https://dvc.org/doc/user-guide/contributing/docs)?

- The PR title should start with "Fix #bugnum: " (if applicable), followed by a
  clear one-line present-tense summary of the changes introduced in the PR. For
  example: "Fix #bugnum: Introduce the first version of the collection editor.".

- Please make sure to mention "Fix #bugnum" (if applicable) somewhere in the
  description of the PR. This enables GitHub to link the PR to the corresponding
  bug.

### CODE REVIEW

**Comment by efiop** [comment]
@jorgeorpinel @shcheklein Why do we need this:

```
The PR title should start with "Fix #bugnum: " (if applicable), followed by a
clear one-line present-tense summary of the changes introduced in the PR. For
example: "Fix #bugnum: Introduce the first version of the collection editor.".
```

github doesn't seem to highlight it anyway, so mentioning it in the PR title doesn't seem very helpful Also, we should probably consider using similar (if not the same rules) for PRs in both dvc and dvc.org, otherwise, it is getting annoying to switch from one to another.

**Comment by efiop** [comment]
Wow, and anyone contributing also has to run `yarn format-check`. That is quite steep for simply wanting to contribute some typo fix. Time to turn on restyled?

**Comment by shcheklein** [comment]
💯 on adding restyled and updating out Github template as we did for the DVC core project :) @jorgeorpinel could create some tickets and/or PRs for this?

**Comment by jorgeorpinel** [comment]
> github doesn't seem to highlight it anyway

Agee, I'm updating it in ac419716.

File: static/docs/command-reference/remote/add.md [hunk 1]
```diff
@@ -250,6 +250,17 @@ $ dvc remote add myremote gs://bucket/path
 $ dvc remote add myremote ssh://user@example.com/path/to/dir
 ```
 
+> **NOTE!** DVC requires SSH and SFTP access to work properly. Please use `ssh`
```
Comment (shcheklein) [inline static/docs/command-reference/remote/add.md]: @jorgeorpinel we should probably change name of the section from Example to Details? and mention above in the main text that there are important remote specific detail below.

**Comment by jorgeorpinel** [comment]
> anyone contributing also has to run `yarn format-check`. That is quite steep for simply wanting to contribute some typo fix

Its run automatically by the pre-commit hook. We'll try to adopt Restyled soon anyway, yes. But I don't think it will replace the pre-commit hook.

**Comment by jorgeorpinel** [comment]
p.s. added issue for restyled.io: #804

**Comment by shcheklein** [comment]
> But I don't think it will replace the hook.

yes, I think there is no intention to replace it

File: static/docs/command-reference/remote/add.md (hunk 1 above)
Comment (jorgeorpinel) [reply static/docs/command-reference/remote/add.md]: How about calling it **Supported types**?

Comment (jorgeorpinel) [reply static/docs/command-reference/remote/add.md]: And yes about the mention in the **Description**.

File: static/docs/command-reference/remote/add.md [hunk 2]
```diff
@@ -250,6 +250,17 @@ $ dvc remote add myremote gs://bucket/path
...
+> and `sftp` CLI commands to ensure that you are able to connect to your remote.
+
+> **NOTE!** Your server's SFTP root might differ from the physical root (see
```
Comment (jorgeorpinel) [inline static/docs/command-reference/remote/add.md]: Should use `**Note!**` instead, for consistency.

File: static/docs/command-reference/remote/add.md (hunk 1 above)
Comment (shcheklein) [reply static/docs/command-reference/remote/add.md]: Yep. sounds good. May be specify `Supported storage types`?

Comment (jorgeorpinel) [reply static/docs/command-reference/remote/add.md]: Updated.

**Comment by jorgeorpinel** [comment]
I reviewed the notes text. Please take a las look guys.

**Comment by jorgeorpinel** [comment]
Cc @efiop (I can't formally request your review since you opened this PR.)

**Comment by shcheklein** [comment]
@jorgeorpinel it looks good! thanks. I think @efiop won't mind if we merge it.

**Comment by efiop** [comment]
@jorgeorpinel Thanks for fixing up the PR! Looks good! 🙂

**Comment by jorgeorpinel** [comment]
np ty

**Pull request merged**